webdriver/tests/bidi/browsing_context/fragment_navigated/fragment_navigated.py

```diff
@@ -0,0 +1,306 @@
+    event = await on_entry
+
+    time_end = await current_time()
+
+    assert_navigation_info(
+        event,
+        {"context": new_tab["context"], "timestamp": int_interval(time_start, time_end)}
+    )
+
+
+async def test_navigation_id(
+    bidi_session, new_tab, url, subscribe_events, wait_for_event
+):
+    await bidi_session.browsing_context.navigate(
+        context=new_tab["context"], url=url(EMPTY_PAGE), wait="complete"
+    )
+
+    await subscribe_events([FRAGMENT_NAVIGATED_EVENT])
+
+    on_frame_navigated = wait_for_event(FRAGMENT_NAVIGATED_EVENT)
+
+    target_url = url(EMPTY_PAGE + '#foo');
+    result = await bidi_session.browsing_context.navigate(
+        context=new_tab["context"], url=target_url, wait="complete")
+
+    recursive_compare(
+        {
+            'context': new_tab["context"],
+            'navigation': result["navigation"],
+            'timestamp': any_int,
+            'url': target_url
+        },
+        await on_frame_navigated,
+    )
+
+
+async def test_url_with_base_tag(bidi_session, subscribe_events, inline, new_tab, wait_for_event):
+    url = inline("""<base href="/relative-path">""")
+    await bidi_session.browsing_context.navigate(context=new_tab["context"], url=url, wait="complete")
+
+    await subscribe_events(events=[FRAGMENT_NAVIGATED_EVENT])
+
+    on_frame_navigated = wait_for_event(FRAGMENT_NAVIGATED_EVENT)
+
+    target_url = url + '#foo';
+    await bidi_session.browsing_context.navigate(context=new_tab["context"], url=target_url, wait="complete")
+
+    recursive_compare(
+        {
+            'context': new_tab["context"],
+            'url': target_url
+        },
+        await on_frame_navigated,
+    )
+
+
+async def test_iframe(
+    bidi_session, new_tab, url, inline, subscribe_events, wait_for_event
+):
+    initial_url = url(EMPTY_PAGE + '#foo')
+    parent_url = inline(f"<iframe src='{initial_url}'></iframe>")
+    await bidi_session.browsing_context.navigate(
+        context=new_tab["context"], url=parent_url, wait="complete"
+    )
+    all_contexts = await bidi_session.browsing_context.get_tree()
+
+    # about:blank + a new tab are top-level contexts.
+    assert len(all_contexts) == 2
+    parent_info = all_contexts[1]
+    assert len(parent_info["children"]) == 1
+    child_info = parent_info["children"][0]
+
+    await subscribe_events([FRAGMENT_NAVIGATED_EVENT])
+
+    on_frame_navigated = wait_for_event(FRAGMENT_NAVIGATED_EVENT)
+
+    target_url = url(EMPTY_PAGE + '#bar')
+    await bidi_session.browsing_context.navigate(
+        context=child_info["context"], url=target_url, wait="complete")
+
+    recursive_compare(
+        {
+            'context': child_info["context"],
+            'timestamp': any_int,
+            'url': target_url
+        },
+        await on_frame_navigated,
+    )
+
+
+@pytest.mark.parametrize(
+    "hash_before, hash_after",
+    [
+        ("", "#foo"),
+        ("#foo", "#bar"),
+        ("#foo", "#foo"),
```

Comment:
Actually I missed that particular test. Given that the fragment is not changing at all, this should trigger the event right?

Same would apply to the other tests as well.

Reply:
did you mean `this should NOT trigger the event`? my current understanding is that it should be let me consult the spec for this.

Reply:
I don't see an early return that would prevent event from firing https://html.spec.whatwg.org/multipage/browsing-the-web.html#scroll-to-fragid

Reply:
As discussed on Matrix @juliandescottes mentioned that internally we also get a `LOCATION_CHANGE_SAME_DOCUMENT` notification when navigating to the same fragment. Also the spec isn't that much clear, so it looks like that it is fine to keep this test.